decidim-comments/app/queries/decidim/comments/metrics/comments_metric_manage.rb

```diff
@@ -0,0 +1,76 @@
+            cumulative_value = results[:cumulative]
+            next if cumulative_value.zero?
+            quantity_value = results[:quantity] || 0
+            space_type, space_id, category_id, related_object_type, related_object_id = key
+            record = Decidim::Metric.find_or_initialize_by(day: @day.to_s, metric_type: @metric_name,
+                                                           participatory_space_type: space_type, participatory_space_id: space_id,
+                                                           organization: @organization, decidim_category_id: category_id,
+                                                           related_object_type: related_object_type, related_object_id: related_object_id)
+            record.assign_attributes(cumulative: cumulative_value, quantity: quantity_value)
+            @registry << record
+          end
+          @registry.each(&:save!)
+          @registry
+        end
+
+        private
+
+        def query
+          return @query if @query
+
+          user_ids = Decidim::User.select(:id).where(organization: @organization).collect(&:id)
+          user_group_ids = Decidim::UserGroup.select(:id).where(organization: @organization).collect(&:id)
+          comments = Decidim::Comments::Comment.includes(:root_commentable).not_hidden
+                                               .where("decidim_comments_comments.created_at <= ?", end_time)
+                                               .where("decidim_comments_comments.decidim_author_id IN (?) OR
+                                                     decidim_comments_comments.decidim_user_group_id IN (?)", user_ids, user_group_ids)
+
+          @query = comments.each_with_object({}) do |comment, query_h|
+            root_commentable = comment.root_commentable
+            return query_h unless root_commentable
+            space_type = ""
+            space_id = ""
+            category_id = ""
+            related_object_type = comment.decidim_root_commentable_type
+            related_object_id = comment.decidim_root_commentable_id
+
+            if root_commentable.is_a? Decidim::Participable
+              space_type = comment.decidim_root_commentable_type
+              space_id = comment.decidim_root_commentable_id
+            elsif root_commentable.is_a? Decidim::Component
+              space_type = root_commentable.participatory_space_type
+              space_id = root_commentable.participatory_space_id
+            elsif comment.root_commentable.is_a? Decidim::HasComponent
+              space_type = root_commentable.component.participatory_space_type
+              space_id = root_commentable.component.participatory_space_id
+            end
+            category_id = root_commentable.category.try(:id) if root_commentable.respond_to?(:category)
+
+            query_h[[space_type, space_id, category_id, related_object_type, related_object_id]] ||= { cumulative: 0, quantity: 0 }
```

Comment:
What's this? So `query_h` is a Hash, where the keys are Arrays? This feels *very* hard to maintain, please avoid it. Maybe wrap this into objects or something.

Reply:
Ok, but this is used in other managers. Do you wnat to change it too?

Reply:
So you mean other metrics managers are using the `@query` instance variable from the `CommentsMetricManage` class? 😕 

(also, the fact that it's a manager, but the class is called `Manage` triggers me a lot)

Reply:
No no. Other manager uses the result of `group` method, that is a `Hash` with multiple keys. Here I'm only trying to emulate that functionality.

Jut to be clear, this manager is the most complex of all of them only because of the commentable relationship.

Reply:
You are correct. `query` method is private for each manager, but I try to keep the same methodology. I'm going to change this for a more readable and understandable code.